mod/peer/gossip/state/state.go

```diff
@@ -78,34 +69,157 @@ type GossipServiceMediator interface {
 
 //AddBlockHandler handles state update in gossip
 func AddBlockHandler(cid string, publisher api.BlockPublisher) {
-	//do nothing
+	publisher.AddWriteHandler(func(blockNum uint64, txID string, namespace string, kvWrite *kvrwset.KVWrite) error {
```

Comment:
I don't think this will work. I think we need to modify the block publisher to allow for channel-specific handlers. The way it is now, when a block event occurs then all handlers are called regardless of channel.

Reply:
block publisher is already channel specific. 
So handler function signatures are updated to include channel IDs.